pkg/api/frontendsettings.go

```diff
@@ -110,9 +111,14 @@ func getFrontendSettingsMap(c *middleware.Context) (map[string]interface{}, erro
 	}
 
 	// add mixed backend data source
-	datasources["-- Mixed --"] = map[string]interface{}{
-		"type": "mixed",
-		"meta": plugins.DataSources["mixed"],
+	for _, ds := range plugins.DataSources {
+		if !ds.Mixed {
+			continue
+		}
+		datasources["-- "+strings.Title(ds.Id)+" --"] = map[string]interface{}{
```

Comment:
I want to reuse query editor of the original datasource. And intercept query by new datasource plugin, and modify query for some condition.

https://github.com/mtanda/grafana-conditional-switch-datasource

I want to change datasource by query time range.
If time range is short enough use A datasource, the time range is long use B datasource.

I'm not sure mixed type datasource is best way.

If normal datasource can wrap other datasource, and reuse other datasource's query editor. I'll create it as normal datasource.